db/src/tx.rs

```diff
@@ -299,10 +341,15 @@ impl<'conn, 'a> Tx<'conn, 'a> {
                     }
 
                     let added = op == OpType::Add;
+                    let reduced = (e, a, attribute, v, added);
+                    if metadata::might_update_metadata(&reduced) {
```

Comment:
This function doesn't use most of the arguments. Can you just pass `e` by value instead?

https://medium.com/@robertgrosse/how-copying-an-int-made-my-code-11-times-faster-f76c66312e0f#.m3xlziaht